client/state/sites/terms/reducer.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import merge from 'lodash/merge';
+import uniq from 'lodash/uniq';
+import get from 'lodash/get';
+import keyBy from 'lodash/keyBy';
+
+/**
+ * Internal dependencies
+ */
+import {
+	TERMS_RECEIVE,
+	DESERIALIZE
+} from 'state/action-types';
+import { isValidStateWithSchema } from 'state/utils';
+import {
+	termsSchema,
+	taxonomiesSchema
+} from './schema';
+
+export function taxonomies( state = {}, action ) {
+	switch ( action.type ) {
+		case TERMS_RECEIVE:
+			const currentTaxonomyTermIds = get( state, [ action.siteId, action.taxonomy ], [] );
+			// This feels a bit odd to me, but uncertain of best way to handle paginated data?
```

Comment:
Especially since we also support a query, I worry we might need to add more complex handling for query results like what we have in `state.posts`.